modules/launcher/src/main/java/org/mule/module/launcher/artifact/LeakCleaner.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.launcher.artifact;
+
+/**
+ * Implementations of this class should take care about objects that may leak memory after application undeployment.
+ */
+public interface LeakCleaner
+{
+
+    void clean();
```

Comment:
I think this method should not throw any exception because if there is an error during memory leak cleaning, that is not an application error per se, and that must be clear in the method's javadoc